sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONOptions.scala

```diff
@@ -51,7 +53,8 @@ private[sql] class JSONOptions(
     parameters.get("allowBackslashEscapingAnyCharacter").map(_.toBoolean).getOrElse(false)
   val compressionCodec = parameters.get("compression").map(CompressionCodecs.getCodecClassName)
   private val parseMode = parameters.getOrElse("mode", "PERMISSIVE")
-  val columnNameOfCorruptRecord = parameters.get("columnNameOfCorruptRecord")
+  val columnNameOfCorruptRecord =
+    parameters.getOrElse("columnNameOfCorruptRecord", sqlConf.columnNameOfCorruptRecord)
```

Comment:
Oh, actually, the reason was not only to remove duplicated logics. Another reason is that we are passing `columnNameOfCorruptRecords` to `InferSchema.infer(...)` and `JacksonParser.parse(...)`


```scala
def infer(
    json: RDD[String],
    columnNameOfCorruptRecords: String,
    configOptions: JSONOptions): StructType = {
  ...
```

```scala
def parse(
    input: RDD[String],
    schema: StructType,
    columnNameOfCorruptRecords: String,
    configOptions: JSONOptions): RDD[InternalRow] = {
  ...
```

although `columnNameOfCorruptRecords` is an option in `JSONOptions`. I would revert this change if it still does not sound good but I just wanted to let you know just in case.